runtime/src/main/java/org/corfudb/runtime/CheckpointLivenessUpdater.java

```diff
@@ -52,34 +59,61 @@ private Table<TableName, ActiveCPStreamMsg, Message> getActiveCheckpointsTable()
                 TableOptions.fromProtoSchema(ActiveCPStreamMsg.class));
     }
 
-    private void addTask() {
-        if (!currentTable.isPresent()) {
-            return;
-        }
-        executorService.execute(() -> {
-            try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
-                ActiveCPStreamMsg currentStatus = (ActiveCPStreamMsg)
-                        txn.getRecord(CompactorMetadataTables.ACTIVE_CHECKPOINTS_TABLE_NAME, currentTable.get()).getPayload();
-                ActiveCPStreamMsg newStatus = ActiveCPStreamMsg.newBuilder()
-                        .setSyncHeartbeat(currentStatus.getSyncHeartbeat() + 1)
-                        .build();
-                // update validity counter for the current table
-                txn.putRecord(activeCheckpointsTable, currentTable.get(), newStatus, null);
-                txn.commit();
-            } catch (Exception e) {
-                log.error("Unable to update liveness for table: {} due to exception: {}", currentTable, e.getStackTrace());
+    private void updateHeartbeat() {
+        TableName table;
+        while (true) {
+            if (currentTableLock.tryLock()) {
+                try {
+                    if (!currentTable.isPresent()) {
+                        return;
+                    }
+                    table = currentTable.get();
+                    break;
+                } finally {
+                    currentTableLock.unlock();
+                }
             }
-        });
+        }
+        try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            ActiveCPStreamMsg currentStatus = (ActiveCPStreamMsg)
+                    txn.getRecord(CompactorMetadataTables.ACTIVE_CHECKPOINTS_TABLE_NAME, table).getPayload();
+            ActiveCPStreamMsg newStatus = ActiveCPStreamMsg.newBuilder()
+                    .setSyncHeartbeat(currentStatus.getSyncHeartbeat() + 1)
+                    .build();
+            // update validity counter for the current table
+            txn.putRecord(activeCheckpointsTable, table, newStatus, null);
+            txn.commit();
+        } catch (Exception e) {
+            log.error("Unable to update liveness for table: {} due to exception: {}", table, e.getStackTrace());
```

Comment:
Following up the previous comment - if it's TransactionAbortedException with cause == CONFLICT, probably log.info is better